### DISCUSSION TITLE
ER Diagram for Snowflake

### DISCUSSION THREAD

**Discussion by archgnu** [dbeaver/dbeaver#14235] [DBeaver Development]
Can someone point me to the location in code of ER Diagram implementation for Snowflake?

I would like to see how these table relationships are implemented?

![ref](https://user-images.githubusercontent.com/90292961/137204370-45e0b1fb-e499-4100-8545-adf94b67f53b.png)

**Comment by LonwoLonwo** [comment]
Hello @archgnu 
We do not have database-specific ERD code. 
The main ERD code you can find in these plugins:
/org.jkiss.dbeaver.erd.model
/org.jkiss.dbeaver.erd.ui

**Reply by archgnu** [reply]
L, thx and sry for bothering, but is there any snowflake specific or general query used for obtaining referenced columns?

**Comment by LonwoLonwo** [comment]
Sorry, no special implementations for Snowflake. We receive information about foreign keys and referenced columns directly from the driver.
Do you think that something goes wrong?
